### DISCUSSION TITLE
توحيد الجهود

### DISCUSSION THREAD

**Discussion by assawalhy** [softvenue/i18n#85] [Ideas]
> <p align=center>السلام عليكم ورحمة الله وبركاته.</p>

----

<div dir=rtl>

وجدت بعض الجهود الأخرى في تعريب المصطلحات التقنية وتساءلت ماذا لو توحدت الجهود...

1. https://www.arabeyes.org/%D8%A7%D9%84%D9%82%D8%A7%D9%85%D9%88%D8%B3_%D8%A7%D9%84%D8%AA%D9%82%D9%86%D9%8A
2. https://www.arabeyes.org/techdict/arabdict.pdf
2. https://techtionary.thinktech.sa/
3. https://dict.linguist.page/
4. الفصل الأول "مصطلحات" من كتاب "[تقنية المعلومات](https://books.google.com.eg/books?id=DaG7DAAAQBAJ&printsec=frontcover&source=gbs_ge_summary_r&cad=0#v=onepage&q&f=false)" لـ د. فيصلُ بن هاشم بن شمس الدين
4. http://www.arabterm.org/index.php?id=3&L=1
5. https://arabicglossary.dubaifuture.ae/ar/browse/
6. https://www.arabdict.com/ar/english-arabic
7. https://www.wikiwand.com/ar/%D9%85%D8%B5%D8%B7%D9%84%D8%AD%D8%A7%D8%AA_%D8%AA%D8%B4%D8%B1%D9%8A%D8%AD%D9%8A%D8%A9
8. https://linguist.fm/

---

فأرى أن تُنشأ منصة تحتوى على المصطلحات وتكون كل هذه الجهود متعاونة لا متنافرة، مع إتاحة مزايا كالتي في قاموس عربآيز مع تجربة مستخدم أفضل وتصميم جديد، فاستعينوا بالله ثم ابدأوا بالتواصل مع أصحاب المحاولات والجهود الأخرى. إن شاء الله أنا مستعد لمساعدتكم في إنشاء المنصة ☺️!

**Comment by assawalhy** [comment]
<div dir=rtl>

صحيح، لديكم [منصة](https://i18n.softvenue.net/) لعرض المصطلحات لكن تنقصها مزايا في عربآيز مثل النقاشات والتصويت على المصطلحات واقتراح ترجمة من خلال المنصة نفسها.

كنت قد [اقترحت مسبقا منصة](https://github.com/mobadarah/tajmeeaton#%D8%AA%D8%B9%D8%B1%D9%8A%D8%A8-%D8%A7%D9%84%D9%85%D8%B7%D9%84%D8%AD%D8%A7%D8%AA-%D8%A7%D9%84%D8%B9%D9%84%D9%85%D9%8A%D8%A9) مشابهة لكن بشكل أشمل، بحيث تكون للمصطلحات في مجالات العلم والحياة. هي نفس المنصة لكن بوضع أقسام مختلفة فيها لكل فرع وقسم، وربما كانت مثل stackoverflow بحيث ينضم المرء لو شاء لمجال من المجالات حتى يُتاح له الاقتراح والتصويت

**Comment by Aissaoui-Ahmed** [comment]
<div dir=rtl >
شكرا أخي @MuhammadSawalhy على اقتراحك الرائع<br>لقد شاهدت تجميعة مجهود ثمين منك اكرر شكري، لك كامل الحرية في التصرف في المنضمة يمكن مراجعة بريدك ونحن نرحب بأي اقتراح تقدمة لقد وجهتك لك دعوة 🚀 
</div>

**Comment by assawalhy** [comment]
<div dir=rtl>

أشكرك وعلى ثقتك وأسأل الله أن أكون عند حسن ظنك :heart: 

إن شاء الله سأعمل على المنصة الشاملة لتعريب المصطلحات بالمزايا التي اقتراحتها، وسأستعين بالمجهودات الأخرى وأكمل عليها. وأظن أنّا يمكننا أن نقسم الجهد بيننا ثم بعد الشروع في المشروع وإصدار نسخة أولية منه أن نتواصل مع أصحاب الإسهامات الأخرى بحيث يضيفوا مصطلحاتهم وأن يساهموا معنا ويكونوا جزءا من القاءمين على تطويره.

هل هذه خطوات سليمة؟! وما رأيك أن تكون معي على @mobadarah لو لديك خبرة بتطوير المواقع سواءً أكانت طرفا خلفيا أو أماميا أو كليهما.

1. أرى أن نبدأ بتحديد المتطلبات وخصائص المنصة بشيء من التفصيل
2. ثم فتح باب النقاش مع الناس على منصات التواصل مثلا
3.  ثم العمل على نسخة أولية ([MVP minimal viable product](https://en.wikipedia.org/wiki/Minimum_viable_product)) من المنصة بعد عمل نموذج أولي prototype ولدي خبرة بهذا
4. ثم البدأ في تطويره وتقسم المهام بيننا
5. كما أرى أن نبحث عن من يشاركنا الأهداف والرُؤى

**Comment by assawalhy** [comment]
<div dir=rtl>

المشكلة أن كلًا منا في دنياه مشغول في طلب العلم والرزق وكثير من المشاريع الغير مكتملة، لكن إن قسمنا العمل وتعاونّا عليه سنصل إن شاء الله وسيقل التسويف والكسل :sweat_smile:

**Comment by Aissaoui-Ahmed** [comment]
> أشكرك وعلى ثقتك وأسأل الله أن أكون عند حسن ظنك ❤️
> 
> إن شاء الله سأعمل على المنصة الشاملة لتعريب المصطلحات بالمزايا التي اقتراحتها، وسأستعين بالمجهودات الأخرى وأكمل عليها. وأظن أنّا يمكننا أن نقسم الجهد بيننا ثم بعد الشروع في المشروع وإصدار نسخة أولية منه أن نتواصل مع أصحاب الإسهامات الأخرى بحيث يضيفوا مصطلحاتهم وأن يساهموا معنا ويكونوا جزءا من القاءمين على تطويره.
> 
> هل هذه خطوات سليمة؟! وما رأيك أن تكون معي على @mobadarah لو لديك خبرة بتطوير المواقع سواءً أكانت طرفا خلفيا أو أماميا أو كليهما.
> 
> 1. أرى أن نبدأ بتحديد المتطلبات وخصائص المنصة بشيء من التفصيل
> 2. ثم فتح باب النقاش مع الناس على منصات التواصل مثلا
> 3. ثم العمل على نسخة أولية ([MVP minimal viable product](https://en.wikipedia.org/wiki/Minimum_viable_product)) من المنصة بعد عمل نموذج أولي prototype ولدي خبرة بهذا
> 4. ثم البدأ في تطويره وتقسم المهام بيننا
> 5. كما أرى أن نبحث عن من يشاركنا الأهداف والرُؤى

<div dir=rtl> إن شاء الله يارب تكون بداية خير<br>
إن اردت البدء فالرجاء الوجه الى <a href="https://github.com/orgs/softvenue/teams/all"> الفريق </a><br>
ملاحظة لقد تم إقتراح قاموس عربآيز من قبل @atefBB هنا #52 وايضا إقترح @AissamYekhlef في تغيير عمل المنصة #70

**Reply by AissamYekhlef** [reply]
أولا شكرا على إهتمامك بعمل هكذا مشروع لدعم اللغة العربية والمجتمع العربي 
ثم أنا في الخدمة من الأن أعمل كمطور full-stack web developer 
إذا احتجت شيء يمكننا التواصل في أي وقت 
شكرا

**Reply by assawalhy** [reply]
<div dir=rtl>

> أولا شكرا على إهتمامك بعمل هكذا مشروع لدعم اللغة العربية والمجتمع العربي 
> ثم أنا في الخدمة من الأن أعمل كمطور full-stack web developer 
> إذا احتجت شيء يمكننا التواصل في أي وقت 
> شكرا 

♥️

أنا أطور بال React وأنت بال Vue 😔،لكن يمكنك استلام الطرف الخلفي وأن تترك لي الأمامي...

**Reply by assawalhy** [reply]
<div dir=rtl>

## المزايا المقترحة للنسخة الأولية

1. صفحة رئيسية تعرف بالمشروع ورؤيتنا وطموحاتنا للمشروع
1. عرض المصطلحات والبحث فيها
1. تصنيف المصطلحات حسب المجال العملي أو الحياتي 
1. إضافة مصطلح جديد من المشرفين على المنصة وهم نحن حتى الآن

## رؤيتي على المدى البعيد

1. جمع وجرد للمصطلحات وتعريباتها من الجهود المبذولة قبلنا بحيث نكمل عليها مع تبيان المصدر 
1. نبحث عن المساهمين ونضيفهم معنا، القادرين على توفير الوقت لإنجاز مهامهم ومن لهم نفس الرؤية وهدفهم نهضة أمتهم
1. اقترح أن يكون الموضع أشبه ب https://stackexchange.com/ خُصص للمصطلحات
    1. به مجتمعات فرعية مثل مجتمع للرياضيات ومجتمع تقني ومجتمع للفيزياء وهكذا...
    1. أن تعطى صلاحيات للمتخصصين في مجلاتهم وأصحاب اللسان من أهل العلم باللغة أو ممن وصل تقييمهم على المنصة وتجاوز القدر المطلوب للتعليق مثلا ثم اقتراح تعريب إن تجاوز تقييمه القدر المطلوب لذلك
    1. تُعطى نقاط للمساهمين على حسب تفاعلهم، وشهادات تقدير ونعرض قائمة بالأكثر تفاعلا في الصفحة الرئيسية
    1. إذا اُرتضي تعريب ونال استحسان الناس وفقا لتصويتهم جعلناه في الأعلى وعرض أولا وعُرض كتعريب للمصلح عند عرض التفاصيل المختصرة للمصطلح
    1. أن يكون مجموعة من المشرفين يتابعون المقتراحات والتعليقات والتعريبات، ويُقصون منها ما وُضع احتيالا وإفسادا، ويوجهون المستخدمين ويجيبوهم (وهم نحن كذلك إن لم يعد عائد من المشروع يتكفل برواتبهم) 
1. يمكننا أن نضع به إعلانات غير مخلة بالدين أو القيم أو تجربة المستخدم وتكون مصدرا وعائدا لتحمل نفقة الاستضافة وقاعدة البيانات التي يجب أن تتحمل الكثير

**Reply by Aissaoui-Ahmed** [reply]
<div dir=rtl>
شرف لي المشاركة في مشاريع داعمة للعربية @mobadarah <br>
مارايك في تحظير قالب (template email) لدعوة واستشارة اصحاب المشاريع واخذ الخبرة <br>
github@softvenue.net مارايكم في هذا البريد او اقتراح اسم اخر <br>
ماريك في التركيز على ترجمة مصطلحات التقنية للبرمجة حاليا من التوجه للخطوة i <br>
 أما الباقي فأنا أوافقك الراي 100%
الرجاء فتح كل اقتراح على حدى (issue) لوضعها في خانات المشاريع <a href=https://github.com/softvenue/i18n/projects>Project</a> والعمل عليها
</div>

**Reply by assawalhy** [reply]
أرسلت لك دعوة للانضمام ل @mobadarah كمالك لها معي (owner)

أظن أن دعوة المهتمين حاليا خطوة مستعجلة، فلنبدأ بإصدار النسخة الأولية وسيظل أمامنا الكثير ليساهم معنا أصحاب المهارات التقنية واللغوية...

**Reply by assawalhy** [reply]
المشروع سيكون أكبر من مجرد ترجمة مصطلحات تقنية إن شاء الله، وسيمتد ليشمل كافة المجالات العلمية والحياتية، ولذا فاقترح أن نبدأ بعمل مستودع جديد أو لربما منظمة organization جديدة على جيتهاب وهذا له عامل آخر وهو ألا نجعل الحاجز النفسي للمساهمين أن هذا العمل سيجني ثماره softvenue معنويا أو ماديا، بل هو عمل للجميع وثماره للنهوض بالمشروع وللماسهمين

هل لو كنا جميعا على mobadarah التي شعارها اللغة العربية وتحتوي مشاريع مسبقة تخص العربية؟

أم ننشيء منظمة جديدة على الجيتهاب، لهذا المشروع وتوابعه من المشاريع؟

**Reply by assawalhy** [reply]
وللإسراع في النسخة الأولية كي لا يتسلل اليأس إلينا ونفقد الأمل في المشروع أرى أن يحدد كل منا ما يمكنه أن يقدم للمشروع ثم نبدأ بتقسيم عمل النسخة الأولية علينا.

ما يمكنني أن أقدمه:
1. الطرف الأمامي frontend (وبإطار عمل React إن احتجنا إطار عمل)
2. النمذجة الأولية prototyping على figma مثلا
3. الطرق الخلفي backend (يمكنني العمل عليه لكن تنقصني الخبرة والعلم العميق، لكني عملت مسبقا وعدلت على خلفيات المواقع)

**Reply by assawalhy** [reply]
@AissamYekhlef
@atefBB 
@Aissaoui-Ahmed

**Reply by atefBB** [reply]
<div dir="rtl">أنا استطيع المساعدة في backend/frontend languages ومراجعة الترجمة إعرابيا ورسما (لا أدّعي أنني من أهل اختصاص اللغة العربية لكنني استطيع المساعدة).</div>

**Comment by atefBB** [comment]
<div dir="rtl">السلام عليكم، جزاكم الله خيرا. أريد أن ألفت نظركم لنقطة مهمة، ألا وهي: أننا نحن المهندسيين والمطوريين لسنا من أهل الاختصاص في اللغة العربية والترجمة - أو معظمنا- لذلك وجب أن يكون ضمن الفريق من أهل الاختصاص من يصحح ويراجع، أو نعمل اﻵن ويتم عرض العمل عليهم. ولكم سديد النظر. وشكرا.</div>

**Reply by assawalhy** [reply]
<div dir=rtl>

وعليكم السلام ورحمة الله وبركاته 

معك حق، واقتراحي أن نوفر المنصة التي تكون وعاءً للمتخصصين وأهل اللغة وأصحاب مَلَكَتها

تفقّد خدمات ومزايا المنصة: https://github.com/softvenue/i18n/discussions/85#discussioncomment-1788577

**Comment by assawalhy** [comment]
<div dir=rtl>

**_بداية العمل_**

- أنشأت ستودعا جديدا: https://github.com/softvenue/taareeb
- بدأت العمل على النموذج الأولي: https://framer.com/projects/Taareeb-frontend--17dOS8srlQwjJPmcjBnX-ffEnz
- إن شاء الله سيكون أول إصدار من منصة "التعريب" بالخصائص التالية
  1. صفحة رئيسية
  2. عرض المصطلحات التي ستكون مضافة مسبقا وليس من خلال طرف خلفي backend
  3.

**Reply by assawalhy** [reply]
سنكمل التواصل إن شاء الله على تيليجرام أو الواتساب أو ما ترتاحون له من تطبيقات المراسلات
